app/src/lib/components/BranchFilesList.svelte

```diff
@@ -27,23 +26,27 @@
 		);
 	}
 
-	// Appending the files in this manner prevents any major freezing when rendering a long list of files
-	// The UI may be slightly sluggish while the files are still rendering in, but its quite a bit of an
-	// improvement over being stuck on a frozen icon
-	async function pushSortedFiles(files: AnyFile[]) {
-		sortedFiles = [];
+	let chunkedFiles: AnyFile[][] = [];
+	let displayedFiles: AnyFile[] = [];
+	let currentDisplayIndex = 0;
 
-		for (let filesChunk of chunk(files, 100)) {
-			sortedFiles = [...sortedFiles, ...filesChunk];
-			await sleep(0);
-		}
+	// Make sure we display when the file list is reset
+	$: {
+		chunkedFiles = chunk(sortLikeFileTree(files), 100);
+		displayedFiles = chunkedFiles[0] || [];
+		currentDisplayIndex = 0;
 	}
```

Comment:
Not sure how you feel here, but I prefer making a function for these things (to make it evident what the change detection depends on).

Reply:
That makes quite a bit of sense as a pattern 👍